### PR TITLE
Define variables at the beginning of scope

### PR DESCRIPTION
**Pull request by alexeyvo**
To allow compilation using MSVC 2010

### CODE REVIEW

File: tests/test-mongoc-uri.c [hunk 1]
```diff
@@ -567,27 +567,19 @@ test_mongoc_uri_read_prefs (void)
...
-
-   tags_dcnyusessd_dcsf_empty = BCON_NEW(
+   bson_t *tags_dcnyusessd_dcsf_empty = BCON_NEW(
       "0", "{", "dc", "ny", "use", "ssd", "}",
       "1", "{", "dc", "sf", "}",
       "2", "{", "}"
    );
-
-   tags_empty = BCON_NEW(
+   bson_t *tags_empty = BCON_NEW(
```
Comment (bjori) [inline tests/test-mongoc-uri.c]: Does this actually change anything?

Comment (alexeyvo) [inline tests/test-mongoc-uri.c]: This code doesn't compile using MSVS 2010 see https://jira.mongodb.org/browse/CDRIVER-981?jql=project%20%3D%20CDRIVER%20AND%20resolution%20%3D%20Unresolved%20ORDER%20BY%20priority%20DESC
MSVS doesn't support c99 standard(only partial support). ansi c standard requires variable definition at the beginning of the scope.

Comment (bjori) [inline tests/test-mongoc-uri.c]: I understand, but I'm not seeing where exactly a variable is defined elsewhere in this change?

All the variables are defined at the top, and then assigned to BCON_NEW(...).

All you changed here was to do the assignment with the variable declaration

Comment (bjori) [inline tests/test-mongoc-uri.c]: Ahh!! I see it now. The *next* line continues the variable declarations. sorry, got it! thanks :)

Comment (alexeyvo) [inline tests/test-mongoc-uri.c]: Yes there are another variable definition below, so if we define and declare this variables at the same time all variables will be declared at the top of scope

**Pull request closed without merging**